Cmdline/Action/KSP.cs

```diff
@@ -373,21 +387,57 @@ private int SetDefaultInstall(DefaultOptions options)
         /// Creates a new fake KSP install after the conditions CKAN tests for valid install directories.
         /// Used for developing and testing purposes.
         /// </summary>
-        private int FakeNewKSPInstall ()
+        private int FakeNewKSPInstall (FakeOptions options)
         {
             // TODO
-            // Create new folder
-            // Currently inside directory in which ckan is started, later choosable
-            // Create fake file buildID.txt
-            // Create fake file readme.txt
+            // buildID64 ?
+            // help ?
+            // DLC support
+            User.RaiseMessage("Test");
+
+            // Parse all options
+            bool DLC = options.dlc;
+            KspVersion version;
+            string installName = options.name;
+            string path;
+            try
+            {
+                version = KspVersion.Parse(options.version);
+                path = Path.GetDirectoryName(options.path);
+            }
+            catch (Exception e)
+            {
+                User.RaiseError(e.ToString());
+                return Exit.ERROR;
+            }
+
 
+            // Try to create the folder structure and the txt files
             try
             {
-                string folder = "_fakeKSP/";
-                System.IO.Directory.CreateDirectory(folder);
-                System.IO.Directory.CreateDirectory(folder + "GameData");
-                System.IO.File.WriteAllText(folder + "buildID.txt", "build id = 023352018.10.17");
-                System.IO.File.WriteAllText(folder + "readme.txt", "Version 1.5.1");
+                // Create a KSP root directory, containing a GameData folder, a buildID.txt and a readme.txt
+                string directory = Path.Combine(path, "fakeKSP");
+                Directory.CreateDirectory(directory);
+                Directory.CreateDirectory(Path.Combine(directory, "GameData"));
+                File.WriteAllText(Path.Combine(directory, "buildID.txt"), String.Format("build id = {0}", version.Build));
+                File.WriteAllText(Path.Combine(directory, "readme.txt"), String.Format("Version {0}", version.ToYalovString()));
+
+                // If a installed DLC should be simulated, we create the needed folder structure and the readme.txt
+                if (DLC && version.CompareTo(KspVersion.Parse("1.4.0")) >= 0)
+                {
+                    Directory.CreateDirectory(Path.Combine(directory, "GameData", "SquadExpansion", "MakingHistory"));
+                    File.WriteAllText(Path.Combine(directory, "GameData", "SquadExpansion", "MakingHistory", "readme.txt"), "Version 1.1.0");
```

Comment:
Let's make the DLC version a command line option. The validation script hard codes it because it has no way of knowing the right version to use, but your new command could be used in other contexts where a desired version is known.